pyramid_hypernova/batch.py

```diff
@@ -139,12 +139,18 @@ def process_responses(self, query, jobs):
         except (HypernovaQueryError, ValueError) as e:
             # the service is unhealthy. fall back to client-side rendering
             __, __, exc_traceback = sys.exc_info()
-
-            error = HypernovaError(
-                type(e).__name__,
-                str(e),
-                [line.rstrip('\n') for line in traceback.format_tb(exc_traceback)],
-            )
+            if hasattr(e, 'error_data'):
+                error = HypernovaError(
+                    name=e.error_data.get('name', None),
+                    message=e.error_data.get('message', None),
+                    stack=e.error_data.get('stack', None),
+                )
+            else:
+                error = HypernovaError(
+                    type(e).__name__,
+                    str(e),
+                    [line.rstrip('\n') for line in traceback.format_tb(exc_traceback)],
+                )
```

Comment:
can we add a couple of comments explaining why we have two cases to handle here? for someone unfamiliar with the code, they might ask: "why are we dynamically checking for this `error_data` attribute? when would it be set and when wouldn't it be set? shouldn't we always know that if it's a `HypernovaQueryError`, it has certain properties on it?"

additionally, they might ask "if we know we have an `error_data` object, why do we need to defensively use `.get` on each of its properties?"